### PR TITLE
Convert identify filter form to Phlex, remove orphaned naming ERB

### LINKED ISSUE #3865: Convert: Identify filter
## ERB Form Conversion

### PR DESCRIPTION
**Pull request by mo-nathan**
## Summary
- Convert the identify observations filter form (navbar clade/region autocompleter) from ERB to a Phlex component (`Components::IdentifyFilterForm`)
- Delete orphaned naming form ERB partials (`_form.erb` and `_fields.erb`) that have no remaining references — `Components::NamingForm` fully replaced them
- Add component test with 4 tests and 31 assertions covering default state, clade filter, region filter, and empty value cases

## Test plan
- [x] `bin/rails test test/components/identify_filter_form_test.rb` — 4 tests, 31 assertions pass
- [x] `bin/rails test test/controllers/observations/identify_controller_test.rb` — integration test passes
- [x] RuboCop clean on all new/modified files
- [x] Manual verification: visit `/observations/identify`, confirm clade/region autocompleter filter works in navbar

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by JoeCohen — COMMENTED** [review]
All good except this:
Clicking the `Clear` button in the navbar clade/region autocompleter filter seems to do nothing. I expected it to clear the filter.

**Review by nimmolo — CHANGES_REQUESTED** [review]
Not sure my suggestions are relevant, but i believe so. Checking `ApplicationForm` now. 

Update: yes, I believe so.

File: app/components/identify_filter_form.rb [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    end
+  end
+
+  def render_search_icon
+    span(class:
+      "glyphicon glyphicon-search form-control-feedback")
+  end
+
+  def render_hidden_field
+    hidden_field(:term_id, data: dual_target("hidden"))
```
[the remark this answers is not in the document]
Comment (mo-nathan) [reply app/components/identify_filter_form.rb]: Good thought\! This is actually a `hidden_field` (invisible `<input type="hidden">`) that stores the autocompleter's selected ID value — it's not a visible field at all. `read_only_field` renders a visible read-only display with a label and form-group wrapper, which would be wrong here. The `hidden_field` call on this line already uses the `ApplicationForm#hidden_field` helper (line 357 of application_form.rb).
